subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/PlatformDependencyDecorator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.jvm;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.ExternalDependency;
+import org.gradle.api.artifacts.ExternalModuleDependency;
+import org.gradle.api.artifacts.MinimalExternalModuleDependency;
+import org.gradle.api.artifacts.ModuleDependency;
+import org.gradle.api.attributes.Category;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderConvertible;
+import org.gradle.internal.deprecation.DeprecationLogger;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+/**
+ * Dependency APIs for using <a href="https://docs.gradle.org/current/userguide/java_platform_plugin.html#java_platform_plugin">Platforms</a> in {@code dependencies} blocks.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface PlatformDependencyDecorator extends DependencyCreators {
+
+    @Inject
+    ObjectFactory getObjectFactory();
+
+    default ExternalModuleDependency platform(CharSequence dependencyNotation) {
+        return platform(getDependencyFactory().create(dependencyNotation));
+    }
+
+    default ExternalModuleDependency platform(@Nullable String group, String name, @Nullable String version) {
+        return platform(getDependencyFactory().create(group, name, version));
+    }
+
+    default <D extends ModuleDependency> D platform(D dependency) {
+        dependency.endorseStrictVersions();
+        dependency.attributes(attributeContainer -> attributeContainer.attribute(Category.CATEGORY_ATTRIBUTE, getObjectFactory().named(Category.class, Category.REGULAR_PLATFORM)));
+        return dependency;
+    }
+
+    default Provider<? extends ExternalModuleDependency> platform(ProviderConvertible<? extends MinimalExternalModuleDependency> dependency) {
```

Comment:
Shouldn't the return type here be `Provider<? extends MinimalExternalModuleDependency>`?

Reply:
I think using EMD is fine here. There's no reason to expose MEMD.

Reply:
But why do we want to hide the more specific type if it's public?  Using MEMD would also keep these methods' return types symmetric with their arguments and consistent across methods.  Is there something about MEMD we want to keep hidden?

Reply:
I think this wasn't super intentional. I think I was just copying and pasting the other one